plugins/modules/omevv/omevv_firmware.py

```diff
@@ -0,0 +1,689 @@
+  maintenance_mode_count_check:
+    description:
+      - Allows to check if any host in cluster is in management mode.
+      - C(true) checks if any host in cluster is in management mode.
+      - C(false) does not check if any host in cluster is in management mode.
+    type: bool
+  reboot_options:
+    description:
+      - Host reboot option for firmware update.
+      - C(FORCEREBOOT) will force reboot the server.
+      - C(SAFEREBOOT) will reboot the server in safe mode.
+      - C(NEXTREBOOT) will not reboot the server.
+    type: str
+    choices: [FORCEREBOOT, SAFEREBOOT, NEXTREBOOT]
+    default: SAFEREBOOT
+  reset_idrac:
+    description:
+      - Whether to reset the iDRAC while performing firmware update.
+      - C(true) resets the iDRAC while performing firmware update.
+      - C(false) does not reset the iDRAC while performing firmware update.
+    type: bool
+  run_now:
+    description:
+      - Whether to run the update job now or later.
+      - C(true) will run the update job instantly.
+      - C(false) will run the update at the specified I(date_time).
+    type: bool
+    required: true
+  targets:
+    description:
+      - The target details for the firmware update operation.
+      - Either I(servicetag) or I(host) is required for the firmware update operation.
+    type: list
+    elements: dict
+    required: true
+    suboptions:
+      firmware_components:
+        description:
+          - List of host firmware components to update.
+          - M(dellemc.openmanage.omevv_firmware_compliance_info) module can
+            be used to fetch the supported firmware components.
+        type: list
+        elements: str
+        required: true
+      host:
+        description:
+          - The IP address or hostname of the host.
+          - I(host) is mutually exclusive with I(servicetag).
+          - M(dellemc.openmanage.omevv_device_info) module can be used to fetch the device
+            information.
+        type: str
+      servicetag:
+        description:
+          - The service tag of the host.
+          - I(servicetag) is mutually exclusive with I(host).
+          - M(dellemc.openmanage.omevv_device_info) module can be used to fetch the
+            device information.
+        type: str
+requirements:
+  - "python >= 3.9.6"
+author:
+  - "Rajshekar P(@rajshekarp87)"
+attributes:
+    check_mode:
+        description: Can run in check_mode and return changed status prediction without modifying target, if not supported the action will be skipped.
+        support: full
+    diff_mode:
+        description: Will return details on what has changed (or possibly needs changing in check_mode), when in diff mode.
+        support: full
+notes:
+    - Run this module from a system that has direct access to Dell OpenManage Enterprise.
+    - This module supports IPv4 and IPv6 addresses.
+"""
+
+EXAMPLES = r"""
+---
+- name: Update the firmware of single component immediately for a specific host
```

Comment:
- name: Immediately update the firmware of a single component for a specific host

Reply:
Incorporated